app/routes/settings/danger-zone.js

```diff
@@ -0,0 +1,11 @@
+import Route from '@ember/routing/route';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+
+export default Route.extend(AuthenticatedRouteMixin, {
+  titleToken() {
+    return this.get('l10n').t('Danger Zone');
+  },
+  model() {
+    return this.get('authManager.currentUser');
```

Comment:
you need to pre-fetch user events and orders here.

```this.get('data.events').length || this.get('data.orders').length```
this is not synchronous. if data.events which basically make a request to users/4/events (example) is used this statement will first return a promise. It will work only when load times are incredibly fast, locally etc. In other cases, the promise will be resolved after this has already executed.